src/server/middleware/systraceMiddleware.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ * 
+ * @flow
+ */
+import type { Middleware, $Request } from 'express';
+
+const fs = require('fs');
+const dedent = require('dedent');
+const logger = require('../../logger');
+
+const systraceMiddleware: Middleware = (req: $Request, res, next) => {
+  if (req.path !== '/systrace') {
+    next();
+    return;
+  }
+
+  const path = `/tmp/react_native_${Date.now()}.json`;
+
+  fs.writeFile(path, req.rawBody, err => {
+    if (err) {
+      next(err);
+      return;
+    }
+
+    logger.debug('Systrace', `/tmp/react_native_${Date.now()}.json`);
+
+    res.end(
+      dedent`
+      Your trace report was saved at ${path}.
+      
+      You can open it with Google Chrome by navigating to 'chrome://tracing'
+      and clicking 'load'.
+
+      The path to the report was printed by Haul in your Terminal so you can
```

Comment:
That's why it's at the end. It's easier to notice what's at the end in a long running server than what's in middle.

Oh, note that this is for printing in the terminal, not for alert message